src/main/java/com/banking/app/bankingApp/service/transactions/TransactionManagementService.java

```diff
@@ -0,0 +1,91 @@
+package com.banking.app.bankingApp.service.transactions;
+
+import com.banking.app.bankingApp.database.transactions.DBTransaction;
+import com.banking.app.bankingApp.database.transactions.TransactionsDatabaseService;
+import com.banking.app.bankingApp.request.transactions.CreateTransaction;
+import com.banking.app.bankingApp.request.transactions.UpdateTransaction;
+import com.banking.app.bankingApp.response.transactions.Transaction;
+import com.banking.app.bankingApp.service.balance.BalanceManagementService;
+
+import javax.naming.InsufficientResourcesException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TransactionManagementService {
+    private TransactionsDatabaseService transactionsDatabaseService;
+    private BalanceManagementService balanceManagementService;
+
+    public TransactionManagementService() {
+        transactionsDatabaseService = new TransactionsDatabaseService();
+        balanceManagementService = new BalanceManagementService();
+    }
+
+    public Transaction addTransaction(CreateTransaction createTransaction) {
+        DBTransaction dbTransaction = transactionsDatabaseService.addDBTransaction(createTransaction);
+
+        if (dbTransaction.getAmount() <= balanceManagementService.getBalance(createTransaction.getSourceId())) {
```

Comment:
In this code, transaction will be written to DB even if there is not enough balance. Did you test this?